rfcs/0006-implementation-api.md

```diff
@@ -0,0 +1,47 @@
+# Storage API
+
+**suRaft** requires a storage interface with the following functions:
+
+- **Write**:
+
+  ```rust
+  fn write(path: &str, buf: &[u8], put_if_absent: bool) -> Result<bool, io::Error>
+  ```
+
+  Writes data to a specified `path`. If `put_if_absent` is `true`, the write occurs only if the path does not already exist. Returns `Ok(true)` if the write is successful and `put_if_absent` conditions are met.
+
+- **Read**:
+
+  ```rust
+  fn read(path: &str) -> Result<Option<Reader>, io::Error>
+  ```
+
+  Reads data from the specified `path`. Returns a `Reader` that provides access to the stored data. Returns `None` if the `path` is absent.
+
+- **List**:
+
+  ```rust
+  fn list(prefix: &str, start_after: &str) -> Result<Vec<String>, io::Error>
+  ```
+
+  Lists all paths in storage that start with the given `prefix`,  beginning after `start_after`.
+
+# Network API
+
+For communication between servers, **suRaft** defines the following network functions:
+
+- **Request Vote**:
+
+  ```rust
+  fn request_vote(&self, req: RequestVote) -> Result<RequestVoteReply, NetError>
```

Comment:
`async fn`? Or the async part will be wrapped outside.

Reply:
Right. It's an `async fn`...